models/iaf_bw_2001_exact.h

```diff
@@ -0,0 +1,529 @@
+ * @note Must have C-linkage for passing to GSL. Internally, it is
+ *       a first-class C++ function, but cannot be a member function
+ *       because of the C-linkage.
+ * @note No point in declaring it inline, since it is called
+ *       through a function pointer.
+ * @param void* Pointer to model neuron instance.
+ **/
+extern "C" inline int iaf_bw_2001_exact_dynamics( double, const double y[], double f[], void* pnode );
+
+// clang-format off
+/* BeginUserDocs: neuron, integrate-and-fire, conductance-based
+
+Short description
++++++++++++++++++
+
+Leaky integrate-and-fire-neuron model with conductance based synapses, and additional NMDA receptors.
+
+Description
++++++++++++
+
+``iaf_bw_2001_exact`` is a leaky integrate-and-fire neuron model with
+
+* an exact implementation of the neuron model described in [1]_.
+* exponential conductance-based AMPA and GABA-synapses
+* NMDA synapses with slow nonlinear dynamics
+* a fixed refractory period
+* no adaptation mechanisms
+
+Neuron and synaptic dynamics
+..................................................
+
+The membrane potential and synaptic variables evolve according to
+
+.. math::
+
+    C_\mathrm{m} \frac{dV(t)}{dt} &= -g_\mathrm{L} (V(t) - V_\mathrm{L}) - I_\mathrm{syn} (t) \\[3ex]
+    I_\mathrm{syn}(t) &= I_\mathrm{AMPA}(t) + I_\mathrm{NMDA}(t) + I_\mathrm{GABA}(t) (t) \\[3ex]
+    I_\mathrm{AMPA} &= (V(t) - V_E)\sum_{j \in \Gamma_\mathrm{ex}}^{N_E}w_jS_{j,\mathrm{AMPA}}(t) \\[3ex]
+    I_\mathrm{NMDA} &= \frac{(V(t) - V_E)}{1+[\mathrm{Mg^{2+}}]\mathrm{exp}(-0.062V(t))/3.57}\sum_{j \in
+    \Gamma_\mathrm{ex}}^{N_E}w_jS_{j,\mathrm{NMDA}}(t) \\[3ex]
+    I_\mathrm{GABA} &= (V(t) - V_I)\sum_{j \in \Gamma_\mathrm{in}}^{N_E}w_jS_{j,\mathrm{GABA}}(t) \\[5ex]
+    \frac{dS_{j,\mathrm{AMPA}}}{dt} &=-\frac{j,S_{\mathrm{AMPA}}}{\tau_\mathrm{AMPA}}+\sum_{k \in \Delta_j} \delta (t - t_j^k) \\[3ex] 
+    \frac{dS_{j,\mathrm{GABA}}}{dt} &= -\frac{S_{j,\mathrm{GABA}}}{\tau_\mathrm{GABA}} + \sum_{k \in \Delta_j} \delta (t - t_j^k) \\[3ex]
+    \frac{dS_{j,\mathrm{NMDA}}}{dt} &= -\frac{S_{j,\mathrm{NMDA}}}{\tau_\mathrm{NMDA,decay}}+ \alpha x_j (1 - S_{j,\mathrm{NMDA}})\\[3ex]
+    \frac{dx_j}{dt} &= -\frac{x_j}{\tau_\mathrm{NMDA,rise}} + \sum_{k \in \Delta_j} \delta (t - t_j^k)
+
+where :math:`\Gamma_\mathrm{ex}` and :math:`\Gamma_\mathrm{in}` are index sets for presynaptic excitatory and inhibitory
+neurons respectively, and :math:`\Delta_j` is an index set for the spike times of neuron :math:`j`.
+
+Since :math:`S_{j,\mathrm{AMPA}}` and :math:`S_{j,\mathrm{GABA}}` are piecewise exponential functions, the sums are also
+a piecewise exponential function, and can be stored in a single synaptic variable each, :math:`S_{\mathrm{AMPA}}` and
+:math:`S_{\mathrm{GABA}}` respectively. The sum over :math:`S_{j,\mathrm{NMDA}}` does not have a simple expression, and
+cannot be simplified. Therefore, for each synapse, we need to integrate separate state variable, which makes the model
```

Comment:
```suggestion
cannot be simplified. Therefore, for each synapse, we need to integrate separate state variables, which makes the model
```
Not sure if variables should be plural or missing an "a" before _separate_ if singular